Use irange in a few places

r-barnes opened the pull request:
Differential Revision: D27573006

pritamdamania87 commented on torch/lib/c10d/ProcessGroupMPI.cpp:
```diff
@@ -717,7 +718,7 @@ c10::intrusive_ptr<ProcessGroup::Work> ProcessGroupMPI::alltoall(
         auto srcFlatDataSplits =
             srcFlatData.split_with_sizes(c10::IntArrayRef(send_lengthsL), 0);
-        for (int i = 0; i < size_; i++) {
+        for(const auto i : c10::irange(size_)) {
```
Curious to know what is the benefit of using irange instead of regular loops? Is this mostly for readability?

r-barnes replied:
I've answered more fully via DM. In short: type safety, const safety, and consistency.

r-barnes committed:
Use irange in a few places (#55325)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/55325

Test Plan: Sandcastle

Differential Revision: D27573006

fbshipit-source-id: eea0a1a28674f14ac68ecad1c656c72fd8a8fabb
